machine_learning/similarity_search.py

```diff
@@ -0,0 +1,113 @@
+"""
+Simularity search is a search algorithm for finding the nearest vector from
+vectors, used in natural language processing.
+In this algorithm, it calculates distance with euclidean distance and
+returns a list containing two data for each vector:
+    1. the nearest vector
+    2. distance between the vector and the nearest vector
+"""
+import math
+from typing import Union
+
+import numpy as np
+
+InputVal = Union[int, float, np.ndarray]
```

Comment:
```suggestion
from collections.abc import Iterable
InputVal = Union[int, float, Iterable]
```

Reply:
Changing euclidean's type hints, deleted things related to types.